v2/protocol/http/options_test.go

```diff
@@ -315,6 +315,93 @@ func TestWithShutdownTimeout(t *testing.T) {
 	}
 }
 
+func TestWithReadTimeout(t *testing.T) {
+	testCases := map[string]struct {
+		t       *Protocol
+		timeout time.Duration
+		want    *Protocol
+		wantErr string
+	}{
+		"valid timeout": {
```

Comment:
nit: please add test for negative values (should break your current implementation)